crates/astria-conductor/src/reader.rs

```diff
@@ -150,65 +145,72 @@ impl Reader {
     }
 
     /// get_new_blocks fetches any new sequencer blocks from Celestia.
-    pub async fn get_new_blocks(&mut self) -> Result<Vec<SequencerBlockData>> {
-        debug!("ReaderCommand::GetNewBlocks");
-        let mut blocks = vec![];
-
+    #[instrument(name = "Reader::get_new_blocks", skip_all)]
+    pub async fn get_new_blocks(&mut self) -> eyre::Result<Vec<SequencerBlockData>> {
         // get the latest celestia block height
         let prev_height = self.curr_block_height;
-        self.curr_block_height = self.celestia_client.get_latest_height().await?;
+        let curr_block_height = self
+            .celestia_client
+            .get_latest_height()
+            .await
+            .wrap_err("failed getting latest height from celestia")?;
+        ensure!(
+            curr_block_height > self.curr_block_height,
+            "no new celestia height"
+        );
```

Comment:
this returns an error? that seems a bit excessive, as this will make a warn log

Reply:
You are right, this would be very noisy but actually normal behavior, so `warn!` is too heavy-handed. I was looking for a way to early return. Will change.

Reply:
reminder on this

Reply:
Changed the early return to `Ok(None)` to address this.

While doing that noticed the conductor kills its event loop if processing failed. Fixed that.